tests/integrations/android_sdk/test_verify_android_sdk.py

```diff
@@ -0,0 +1,281 @@
+
+    # The returned SDK has the expected root path.
+    assert str(sdk.root_path) == str(root_path)
+
+
+def test_invalid_user_provided_sdk(mock_command, tmp_path):
+    "If the user specifies an invalid ANDROID_SDK_ROOT, it is ignored"
+
+    # Create `sdkmanager` and the license file
+    # for the *briefcase* SDK.
+    root_path = tmp_path / ".briefcase" / "tools" / "android_sdk"
+    tools_bin = root_path / "tools" / "bin"
+    tools_bin.mkdir(parents=True, mode=0o755)
+    (tools_bin / "sdkmanager").touch(mode=0o755)
+    licenses = root_path / "licenses"
+    licenses.mkdir(parents=True, mode=0o755)
+    (licenses / "android-sdk-license").touch()
+
+    # Set the environment to specify an ANDROID_SDK_ROOT that doesn't exist
+    mock_command.os.environ = {
+        'ANDROID_SDK_ROOT': str(tmp_path / "other_sdk")
+    }
+
+    # Expect verify_android_sdk() to succeed
+    sdk = verify_android_sdk(mock_command)
+
+    # No calls to download, run or unpack anything.
+    mock_command.download_url.assert_not_called()
+    mock_command.subprocess.run.assert_not_called()
+    mock_command.subprocess.check_output.assert_not_called()
+    mock_command.shutil.unpack_archive.assert_not_called()
+
+    # The returned SDK has the expected root path.
+    assert sdk.root_path == root_path
+
+
+@pytest.mark.parametrize("host_os", ("ArbitraryNotWindows", "Windows"))
+def test_download_sdk(mock_command, tmp_path, host_os):
+    "If an SDK is not available, one will be downloaded"
+    root_path = tmp_path / ".briefcase" / "tools" / "android_sdk"
+
+    # Mock-out `host_os` so we only do our permission check on non-Windows.
+    mock_command.host_os = host_os
+
+    # The download will produce a cached file
+    cache_file = MagicMock()
+    cache_file.__str__.return_value = "/path/to/download.zip"
+    mock_command.download_url.return_value = cache_file
+
+    # Create a file that would have been created by unpacking the archive
+    example_tool = root_path / "tools" / "bin" / "exampletool"
+    example_tool.parent.mkdir(parents=True)
+    example_tool.touch(0o644)
+
+    # Set up a side effect for accepting the license
+    mock_command.subprocess.run.side_effect = accept_license(root_path)
+
+    # Call `verify_android_sdk()`
+    sdk = verify_android_sdk(mock_command)
+
+    # Validate that the SDK was downloaded and unpacked
+    url = "https://dl.google.com/android/repository/sdk-tools-{host_os}-4333796.zip".format(
+        host_os=host_os.lower()
+    )
+    mock_command.download_url.assert_called_once_with(
+        url=url,
+        download_path=mock_command.dot_briefcase_path / "tools",
+    )
+    mock_command.shutil.unpack_archive.assert_called_once_with(
+        "/path/to/download.zip",
+        extract_dir=str(root_path)
+    )
+
+    # The cached file will be deleeted
+    cache_file.unlink.assert_called_once_with()
+
+    # On non-Windows, ensure the mocked binary was made executabley
```

Comment:
`executabley` is like `chocolatey`, which is very Windows-y ;) https://chocolatey.org/

Anyway, you probably want to delete the letter `y`. :)

**Also** I'd say: "ensure the **unpacked** binary" rather than ensure the "**mocked** binary"